src/java/org/apache/cassandra/db/virtual/AccordDebugKeyspace.java

```diff
@@ -577,6 +603,354 @@ public DataSet data()
+                  .column("save_status", Objects.toString(command.saveStatus()))
+                  .column("route", Objects.toString(command.route()))
+                  .column("participants_owns", toStr(command, StoreParticipants::owns, StoreParticipants::stillOwns))
+                  .column("participants_touches", toStr(command, StoreParticipants::touches, StoreParticipants::stillTouches))
+                  .column("participants_has_touched", Objects.toString(command.participants().hasTouched()))
+                  .column("participants_executes", toStr(command, StoreParticipants::executes, StoreParticipants::stillExecutes))
+                  .column("participants_waits_on", toStr(command, StoreParticipants::waitsOn, StoreParticipants::stillWaitsOn))
+                  .column("durability", Objects.toString(command.durability()))
+                  .column("execute_at", Objects.toString(command.executeAt()))
+                  .column("txn", Objects.toString(command.partialTxn()))
+                  .column("deps", Objects.toString(command.partialDeps()))
+                  .column("waiting_on", Objects.toString(command.waitingOn()))
+                  .column("writes", Objects.toString(command.writes()))
+                  .column("result", Objects.toString(command.result()));
+            }
+
+            return ds.getPartition(partitionKey);
+        }
+
+        @Override
+        public Iterator<Partition> getPartitions(DataRange range)
+        {
+            throw new UnsupportedOperationException();
+        }
+    }
+
+    public static final class JournalTable extends AbstractVirtualTable implements AbstractVirtualTable.DataSet
+    {
+        static class Entry
+        {
+            final int commandStoreId;
+            final long segment;
+            final int position;
+            final CommandChange.Builder builder;
+
+            Entry(int commandStoreId, long segment, int position, CommandChange.Builder builder)
+            {
+                this.commandStoreId = commandStoreId;
+                this.segment = segment;
+                this.position = position;
+                this.builder = builder;
+            }
+        }
+
+        private JournalTable()
+        {
+            super(parse(VIRTUAL_ACCORD_DEBUG, JOURNAL,
+                        "Accord per-CommandStore Journal State",
+                        "CREATE TABLE %s (\n" +
+                        "  txn_id text,\n" +
+                        "  command_store_id int,\n" +
+                        "  segment bigint,\n" +
+                        "  segment_position int,\n" +
+                        "  save_status text,\n" +
+                        "  route text,\n" +
+                        "  durability text,\n" +
+                        "  execute_at text,\n" +
+                        "  executes_at_least text,\n" +
+                        "  txn text,\n" +
+                        "  deps text,\n" +
```

Comment:
waiting on here?

Reply:
We cannot easily report waiting_on from the journal because we don't necessarily have the deps that coincide without potentially reporting information from another segment. I agree it would be nice to report it at some point, but requires more thought than I have time for right now.